gator/fragments.py

```diff
@@ -94,14 +98,14 @@ def count_words(contents):
     # count all of the words in each paragraph
     word_counts = []
     for para in paragraphs:
-        # split the string by whitespace (newlines, spaces, etc.) and punctuation
-        words = re.sub(r"[!\"#$%&()*+,\./:;\<=\>\?\@\[\]\^`\{\|\}]", " ", para).split()
+        # split the string by whitespace (e.g., newlines or spaces) and punctuation
+        words = re.sub(WHITESPACE_RE, constants.markers.Space, para).split()
         word_counts.append(len(words))
     # return the minimum number of words across all paragraphs
     if word_counts:
         return min(word_counts)
```

Comment:
`count_words` and `count_total_words` only differ by this line. Perhaps a new method to simplify both of these could be created, reducing duplication?

Reply:
Okay, that makes sense, I can look into this soon. Thanks!

Reply:
Side note: I agree that this should be fixed and I'm glad to do it in this PR. Notably, this is not related the the purpose of this PR. I will go ahead and fix it here, though. Okay, @Michionlion?

Reply:
Note that this refactoring resulted in making it easier to also refactor several other functions in the system, which then percolated all the way back to the invoke module!